bumping all ApplicationInsights packages to 2.21.0

brettsam opened the pull request.

brettsam committed:
bumping all ApplicationInsights packages to 2.21.0

kshyju reviewed (approved):
Need to update release notes as well?

brettsam committed:
update release notes

brettsam commented:
> Need to update release notes as well?

Yup, good call. Done in latest commit.

brettsam committed:
updating depsjson

brettsam commented:
For reference, here's the depsjson failure before i updated it:

```
IMPORTANT: The dependencies in WebHost have changed and MUST be reviewed before proceeding. Please follow up with brettsam, fabiocav or mathewc for approval.

Previous file: D:\a_work\1\s\test\WebJobs.Script.Tests\bin\Debug\net6.0\Microsoft.Azure.WebJobs.Script.WebHost.deps.json
New file:      D:\a_work\1\s\src\WebJobs.Script.WebHost\bin\Debug\net6.0\Microsoft.Azure.WebJobs.Script.WebHost.deps.json

Changed:
- Microsoft.AI.DependencyCollector.dll: 2.20.0.103/2.20.0.103 -> 2.21.0.429/2.21.0.429
- Microsoft.AI.EventCounterCollector.dll: 2.20.0.103/2.20.0.103 -> 2.21.0.429/2.21.0.429
- Microsoft.AI.PerfCounterCollector.dll: 2.20.0.103/2.20.0.103 -> 2.21.0.429/2.21.0.429
- Microsoft.AI.ServerTelemetryChannel.dll: 2.20.0.103/2.20.0.103 -> 2.21.0.429/2.21.0.429
- Microsoft.AI.WindowsServer.dll: 2.20.0.103/2.20.0.103 -> 2.21.0.429/2.21.0.429
- Microsoft.ApplicationInsights.AspNetCore.dll: 2.20.0.103/2.20.0.103 -> 2.21.0.429/2.21.0.429
- Microsoft.ApplicationInsights.dll: 2.20.0.103/2.20.0.103 -> 2.21.0.429/2.21.0.429
- Microsoft.Extensions.Logging.ApplicationInsights.dll: 2.20.0.103/2.20.0.103 -> 2.21.0.429/2.21.0.429

Removed:

Added:
```
